### PR TITLE
Fixes #20956 - Don't pass class constants to AR conditions

### PR DESCRIPTION
**Pull request by mmoll**

### CODE REVIEW

**Comment by mmoll** [comment]
taken from Dominic's rails51 branch

File: app/models/host/base.rb [hunk 1]
```diff
@@ -181,7 +181,7 @@ def set_interfaces(parser)
 
       ipmi = parser.ipmi_interface
       if ipmi.present?
-        existing = self.interfaces.where(:mac => ipmi[:macaddress], :type => Nic::BMC).first
+        existing = self.interfaces.where(:mac => ipmi[:macaddress], :type => Nic::BMC.name).first
```
Comment (tbrisker) [inline app/models/host/base.rb:184]: not that it makes much difference, but why not just `"Nic::BMC"`?

**Review by tbrisker — COMMENTED** [review]
Looks good, I just hope we aren't missing any other locations here.

File: app/models/host/base.rb (hunk 1 above)
Comment (mmoll) [reply app/models/host/base.rb:184]: I guess it was done for the sake of diff-minimizing, but IMHO using the Constant instead of a string is a tad nicer.

Comment (tbrisker) [reply app/models/host/base.rb:184]: most other places currently pass the string, but WFM, not worth the effort..

**Comment by tbrisker** [comment]
Thanks @mmoll and @domcleal !

**Pull request merged**